app/controllers/interactions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class InteractionsController < ApplicationController
+  def create
+    interaction = Interaction.new(contact_id: params[:interaction][:contact_id], interaction_date: params[:interaction][:interaction_date])
+    interaction.user = current_user
+
+    medium = Medium.find_or_create_by(name: params[:interaction][:medium], user: current_user)
+
+    interaction.medium = medium
+
+    unless params[:interaction][:notes_attributes].values.first[:text].empty?
+      interaction.notes.build(params[:interaction][:notes_attributes].values.first)
+    end
+
+    if interaction.save
+      redirect_to contact_path(interaction.contact_id)
+    else
+      render :new
+    end
+  end
+end
```

Comment:
This method is HUUUUGEEE 🐻 Let's try and break it down into a few smaller methods, maybe 🍰

You could have a `set_contact`, `interaction_params` and `find_or_create_medium` method for instance 🐌

Another idea would be to have some kind of error handling, too, like providing a notice on successful creation or rendering `:new` with an unprocessable entity status if the save fails. 🛟